### PR TITLE
Add kubernetes version in OKE node pool spec

### LINKED ISSUE #162: OKE: Introduce Kubernetes version in OKEManagedMachinePoolSpec
**What would you like to be added**:
OKE: Introduce Kubernetes version in OKEManagedMachinePoolSpec. This will allow atomic updates of node pool.

**Why is this needed**:

### PR DESCRIPTION
**Pull request by shyamradhakrishnan**
**What this PR does / why we need it**:

Add kubernetes version to ocimanagedmachinepool spec. Also fix an issue where there was an unwanted node pool upgrade.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #162

### CODE REVIEW

**Comment by shyamradhakrishnan** [comment]
unit tests

```
Using cached envtest tools from /Users/shyaradh/oke/src/github.com/oracle/cluster-api-provider-oci/testbin
setting up env vars
?   	github.com/oracle/cluster-api-provider-oci	[no test files]
ok  	github.com/oracle/cluster-api-provider-oci/api/v1beta1	1.785s	coverage: 17.9% of statements
?   	github.com/oracle/cluster-api-provider-oci/cloud/config	[no test files]
ok  	github.com/oracle/cluster-api-provider-oci/cloud/ociutil	0.607s	coverage: 25.0% of statements
ok  	github.com/oracle/cluster-api-provider-oci/cloud/scope	166.737s	coverage: 73.2% of statements
?   	github.com/oracle/cluster-api-provider-oci/cloud/scope/mocks	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/base	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/base/mock_base	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/compute	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/compute/mock_compute	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/computemanagement	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/containerengine	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/containerengine/mock_containerengine	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/identity	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/identity/mock_identity	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/networkloadbalancer	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/networkloadbalancer/mock_nlb	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/vcn	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/vcn/mock_vcn	[no test files]
ok  	github.com/oracle/cluster-api-provider-oci/controllers	28.239s	coverage: 69.0% of statements
ok  	github.com/oracle/cluster-api-provider-oci/exp/api/v1beta1	2.162s	coverage: 10.1% of statements
ok  	github.com/oracle/cluster-api-provider-oci/exp/controllers	1.560s	coverage: 43.4% of statements
?   	github.com/oracle/cluster-api-provider-oci/feature	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/version	[no test files]
```

File: cloud/scope/managed_control_plane.go [hunk 1]
```diff
@@ -517,7 +518,16 @@ func (s *ManagedControlPlaneScope) UpdateControlPlane(ctx context.Context, okeCl
...
+		// printing json specs will help debug problems when there are spurious/unwanted updates
+		jsonSpec, err := json.Marshal(*spec)
+		if err != nil {
+			return false, err
+		}
+		jsonActual, err := json.Marshal(*actual)
+		if err != nil {
+			return false, err
+		}
+		s.Logger.Info("Control plane", "spec", jsonSpec, "actual", jsonActual)
```
Comment (joekr) [inline cloud/scope/managed_control_plane.go:530]: Should we be doing this for all of our updates?

File: exp/api/v1beta1/ocimanagedmachinepool_types.go [hunk 2]
```diff
@@ -21,6 +21,9 @@ type OCIManagedMachinePoolSpec struct {
 	// +optional
 	ProviderID *string `json:"providerID,omitempty"`
 
+	// Version represents the version of the OKE node pool.
```
Comment (joekr) [inline exp/api/v1beta1/ocimanagedmachinepool_types.go:24]:
```suggestion
	// Version represents the kubernetes version of the OKE node pool.
```

I'm not 100% sure my change is correct, but I wanted to make sure it was as clear that the version is for kubernetes or something else. The way it read to me was some OKE node pool version and I'm not sure what that is.

File: templates/cluster-template-managed-private.yaml [hunk 3]
```diff
@@ -78,8 +78,13 @@ metadata:
   name: ${CLUSTER_NAME}-mp-0
   namespace: default
 spec:
+  version: "${KUBERNETES_VERSION}"
```
Comment (joekr) [inline templates/cluster-template-managed-private.yaml:81]: thought: what about `kubeVersion` or `kubernetesVersion` or something? I don't love the length of it but it is much more clear that way. 🤷

File: templates/cluster-template-managed.yaml [hunk 4]
```diff
@@ -59,6 +59,7 @@ metadata:
   name: ${CLUSTER_NAME}-mp-0
   namespace: default
 spec:
+  version: "${KUBERNETES_VERSION}"
```
Comment (joekr) [inline templates/cluster-template-managed.yaml:62]: I was thinking same for this? `kubeVersion` or `kubernetesVersion` 🤷

**Review by joekr — APPROVED** [review]

**Pull request merged**